src/Conversion/ONNXToTOSA/Tensor/Reshape.cpp

```diff
@@ -0,0 +1,68 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//===---------------- Elementwise.cpp - Elementwise Op --------------------===//
+//
+// Copyright (c) 2022 Advanced Micro Devices, Inc.
+//
+// =============================================================================
+//
+// This file lowers ONNX ReshapeOp to TOSA dialect.
+//
+//===----------------------------------------------------------------------===//
+
+#include "mlir/Dialect/Tosa/IR/TosaOps.h"
+#include "mlir/IR/BuiltinAttributes.h"
+#include "mlir/IR/BuiltinTypes.h"
+#include "mlir/IR/TypeUtilities.h"
+#include "src/Conversion/ONNXToTOSA/ONNXToTOSACommon.hpp"
+#include "src/Conversion/ONNXToTOSA/ONNXToTOSALegalizeUtils.hpp"
+#include "src/Dialect/ONNX/ONNXOps.hpp"
+#include "llvm/ADT/SmallVector.h"
+
+using namespace mlir;
+
+namespace onnx_mlir {
+
+namespace {
+
+class ONNXReshapeLoweringToTOSA : public OpConversionPattern<ONNXReshapeOp> {
+public:
+  using OpConversionPattern::OpConversionPattern;
+  using OpAdaptor = typename ONNXReshapeOp::Adaptor;
+  LogicalResult matchAndRewrite(ONNXReshapeOp op, OpAdaptor adaptor,
+      ConversionPatternRewriter &rewriter) const override {
+
```

Comment:
I don't see any handling of the optional [`allowzero`](https://github.com/onnx/onnx/blob/main/docs/Operators.md#reshape) attribute. The conversion should fail if it has a value that is not implemented. In fact I don't see tosa saying anything that lets me know what to expect from the ONNX "a dimension could also be 0". Can you determine anything from the https://www.mlplatform.org/tosa/tosa_spec.html#_reshape `Operation Function`?

If not, I suggest checking for a zero shape and failing. During the upstream review we may get more help.